src/@utils/provider.ts

```diff
@@ -56,3 +53,13 @@ export async function getFileInfo(
     }
   }
 }
+
+export async function isValidProvider(url: string): Promise<boolean> {
```

Comment:
this method seems overkill as `await ProviderInstance.isValidProvider(url, fetch)` can simply be called in the `FormFields/Provider` part. Likewise the PR title/final commit message is super confusing as original task and todo was meant to remove that dependency and hit provider API directly, but this PR simply switches methods for it, so no dependency is removed. All titles should reflect that